File: runtime/flutter/widgets/sliverSafeArea.ts
import {RuntimeBaseClass} from "../../runtimeBaseClass";
import {Type} from "../../dart/core/type";

import {EdgeInsets} from "./../painting/edgeInsets";
import {Widget} from "./../widget";
import {StatelessWidget} from "./statelessWidget";

export interface SliverSafeAreaProps {
    left?: boolean | undefined;
    top?: boolean | undefined;
    right?: boolean | undefined;
    bottom?: boolean | undefined;
    minimum?: EdgeInsets | undefined;
    sliver: Widget;
}

declare const flutter: {
    widgets: {
        sliverSafeArea: (this: void, props: SliverSafeAreaProps) => SliverSafeArea;
    };
};

export class SliverSafeArea extends StatelessWidget implements RuntimeBaseClass 
{
    public readonly internalRuntimeType = new Type(SliverSafeArea);
    public props: SliverSafeAreaProps;
    public constructor(props: SliverSafeAreaProps) 
    {
        super();
        this.props = props;

        if (this.props.left === undefined) 
        {
            this.props.left = true;
        }
        if (this.props.top === undefined) 
        {
            this.props.top = true;
        }
        if (this.props.right === undefined) 
        {
            this.props.right = true;
        }
        if (this.props.bottom === undefined) 
        {
            this.props.bottom = true;
        }
        if (this.props.minimum === undefined) 
        {
            this.props.minimum = EdgeInsets.zero();
        }
    }

    public build(): Widget 
    {
        return flutter.widgets.sliverSafeArea(this.props);
    }
}

File: lib/cfr/builtins/libs/flutter/widgets/sliverSafeArea.dart
import 'package:hydro_sdk/cfr/builtins/boxing/unboxers.dart';
import 'package:hydro_sdk/hydroState.dart';
import 'package:hydro_sdk/cfr/vm/context.dart';
import 'package:hydro_sdk/cfr/vm/table.dart';
import 'package:flutter/material.dart';

void loadSliverSafeArea(
    {@required HydroState luaState, @required HydroTable table}) {
  table["sliverSafeArea"] = makeLuaDartFunc(func: (List<dynamic> args) {
    return [
      SliverSafeArea(
        left: args[0]["left"],
        top: args[0]["top"],
        right: args[0]["right"],
        bottom: args[0]["bottom"],
        minimum: maybeUnBoxAndBuildArgument<EdgeInsets>(args[0]["minimum"],
            parentState: luaState),
        sliver: maybeUnBoxAndBuildArgument<Widget>(args[0]["sliver"],
            parentState: luaState),
      )
    ];
  });
}
